src/js/components/Button/StyledButtonKind.js

```diff
@@ -113,11 +113,18 @@ const getPath = (theme, path) => {
   return obj;
 };
 
-const adjustPadStyle = (pad, width) => {
-  const offset = parseMetricToNum(width);
+const adjustPadStyle = (pad, width, theme) => {
+  const intelligentPad = theme?.button?.intelligentPad;
+  if (intelligentPad === true) {
+    const offset = parseMetricToNum(width);
+    return css`
+      padding: ${Math.max(parseMetricToNum(pad.vertical) - offset, 0)}px
+        ${Math.max(parseMetricToNum(pad.horizontal) - offset, 0)}px;
+    `;
+  }
+  // If intelligentPad is false return padding without adjustment
   return css`
-    padding: ${Math.max(parseMetricToNum(pad.vertical) - offset, 0)}px
-      ${Math.max(parseMetricToNum(pad.horizontal) - offset, 0)}px;
+    padding: ${pad.vertical} ${pad.horizontal};
```

Comment:
Do we need to add `px`? I saw on the css snippet above we added it there so wasn't sure if we also needed it here
`padding: ${pad.vertical}px ${pad.horizontal}px;`

Reply:
I think in the previous snippet we've parsed metric to num, so we're adding 'px' back. That said, Brittany can you confirm what we have documented in terms of can "pad" be a px and/or t-shirt size? If so, just making sure the t-shirt size works.

Reply:
correct 
``` previous snippet we've parsed metric to num, so we're adding 'px' back. ```

however that said this does not support `t-shirt` sizes however this is a current issue

Reply:
Ok good to know, thanks